core/src/autogluon/core/models/ensemble/stacker_ensemble_model.py

```diff
@@ -217,6 +217,8 @@ def _add_stack_to_feature_metadata(self):
                 self.feature_metadata = stacker_feature_metadata
             else:
                 # FIXME: This is a hack, stack feature special types should be already present in feature_metadata, not added here
+                existing_features = self.feature_metadata.get_features()
                 existing_stack_features = self.feature_metadata.get_features(required_special_types=[S_STACK])
+                stacker_feature_metadata = stacker_feature_metadata.keep_features([feature for feature in existing_features if feature in type_map_raw])
```

Comment:
could you explain why L222 needed for feature-selection, but was not needed before?

Reply:
I believe it was because AutoGluon auto-adds all base learner outputs features into stacker_feature_metadata, and if I didn't do this the program would crash when FeatureSelector prunes a base learner output feature.